Assets/Scripts/SettingsInfo.cs

```diff
@@ -0,0 +1,253 @@
+    {
+        // Check limits for Audio
+        Mathf.Clamp(settings.masterVolume, 0f, 1f);
+        Mathf.Clamp(settings.musicVolume, 0f, 1f);
+        Mathf.Clamp(settings.sfxVolume, 0f, 1f);
+
+        // Check limits for Graphics
+        Math.Clamp(settings.resolutionPresetIndex, 0, resolutions.Length - 1);
+        Math.Clamp(settings.qualityPresetIndex, 0, qualityNames.Length - 1);
+        Math.Clamp(settings.displayMode, 0, 3);
+
+        // Check lower limits for Gameplay
+        if (settings.sensScale <= 0f)
+            settings.sensScale = 0.01f;
+
+        if (settings.playerFOV < 20f)
+            settings.playerFOV = 20f;
+
+        if (settings.zoomFOV < 5f)
+            settings.zoomFOV = 5f;
+    }
+
+    private void CreateDefaultFile()
+    {
+        string jsonData = JsonUtility.ToJson(settings);
+        File.WriteAllText(FilePath, jsonData);
+    }
+
+    public void SaveSettingsData()
+    {
+        CreateDefaultFile();
+    }
+
+    public void ApplyAllSettings()
+    {
+        SetResolutionLevel(settings.resolutionPresetIndex);
+        SetDisplayMode(settings.resolutionPresetIndex);
+        SetQualityLevel(settings.qualityPresetIndex);
+        SetMasterVolume(settings.masterVolume);
+        SetMusicVolume(settings.musicVolume);
+    }
+    #endregion
+    #region Audio methods
+    private float LinearToLogarithmicVolume(float volume)
+    {
+        return 20 * (Mathf.Log10(10 * Mathf.Max(volume, .0001f)) - 1);
+    }
+
+    public void SetMasterVolume(float volume)
+    {
+        settings.masterVolume = volume;
+        mainAudioMixer.SetFloat(audioGroupMaster, LinearToLogarithmicVolume(settings.masterVolume));
+    }
+
+    public void SetMusicVolume(float volume)
+    {
+        settings.musicVolume = volume;
+        mainAudioMixer.SetFloat(audioGroupMusic, LinearToLogarithmicVolume(settings.musicVolume) + maxVolumeMusic);
+    }
+
+    public void SetSFXVolume(float volume)
+    {
+        settings.sfxVolume = volume;
+        mainAudioMixer.SetFloat(audioGroupSFX, LinearToLogarithmicVolume(settings.sfxVolume) + maxVolumeSFX);
+    }
+    #endregion
+    #region Graphic methods
+    public void SetQualityLevel(int index)
+    {
+        settings.qualityPresetIndex = index;
+        QualitySettings.SetQualityLevel(qualityNames.Length - index - 1);
+    }
+
+    public void SetResolutionLevel(int index)
+    {
+        settings.resolutionPresetIndex = index;
+        var resolution = resolutions[index];
+        Screen.SetResolution(resolution.width, resolution.height, true);
+    }
+
+    public void SetDisplayMode(int index)
+    {
+        settings.displayMode = index;
+        Screen.fullScreenMode = (FullScreenMode)index;
+    }
+    #endregion
+    #region Gameplay methods
+    public float ClampSensValue(float value)
+    {
+        return Mathf.Clamp(value, lowerSensLimit, upperSensLimit);
+    }
+    public void SetSensMultiplier(float scale)
+    {
+        settings.sensScale = scale;
```

Comment:
Huh. I'm not sure why you're somehow able to mutate a struct here. This *works*, but the fact that something like this actually works seems rather fishy – structs are supposed to be immutable :thinking: